### PR TITLE
Add JSON OBJECT Operator 

### PR DESCRIPTION
**Pull request by divyansh-sh**

### CODE REVIEW

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 1]
```diff
@@ -12942,4 +12943,47 @@ private String toSqlWithBloat(RelNode root, int bloat) {
     assertThat(toSql(root, DatabaseProduct.CALCITE.getDialect()), isLinux(expectedSql));
     assertThat(toSql(root, DatabaseProduct.BIG_QUERY.getDialect()), isLinux(expectedBiqQuery));
   }
+
+  @Test public void testJSON_OBJECT() {
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:12947]: make it testJsonObjectFunction

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 2]
```diff
@@ -12942,4 +12943,47 @@ private String toSqlWithBloat(RelNode root, int bloat) {
...
+    }
+
+    final RexNode jsonNode = builder.call(SqlLibraryOperators.JSON_OBJECT, operands);
+
+    final RelNode doyRoot = builder
+        .scan("EMP")
+        .project(jsonNode)
+        .build();
+
+    final String expectedMONBiqQuery = "SELECT JSON_OBJECT('Surname', 'Mark', 'Age', '30', " +
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:12968]: change this to `expectedBiqQuery`

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java [hunk 3]
```diff
@@ -12942,4 +12943,47 @@ private String toSqlWithBloat(RelNode root, int bloat) {
...
+    final RelNode root = builder
+        .scan("EMP")
+        .project(builder.alias(toParseJson, "FD"))
+        .build();
+    final String expectedSql = "SELECT PARSE_JSON(NULL) AS \"FD\"\n"
+        + "FROM \"scott\".\"EMP\"";
+    final String expectedBiqQuery = "SELECT PARSE_JSON(NULL) AS FD\n"
+        + "FROM scott.EMP";
+    assertThat(toSql(root, DatabaseProduct.BIG_QUERY.getDialect()), isLinux(expectedBiqQuery));
+  }
```
Comment (nosheenSheikh02) [inline core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:12988]: Add test case with values and column

**Review by nosheenSheikh02 — CHANGES_REQUESTED** [review]

**Comment by nosheenSheikh02** [comment]
Please resolve conflicts

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 3 above)
Comment (SaiTejaVenkatapuram) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java:12988]: Added the test case

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 2 above)
Comment (SaiTejaVenkatapuram) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: changed expectedMONBiqQuery to expectedBiqQuery

File: core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java (hunk 1 above)
Comment (SaiTejaVenkatapuram) [reply core/src/test/java/org/apache/calcite/rel/rel2sql/RelToSqlConverterTest.java]: changed testJSON_OBJECT to testJsonObjectFunction

**Review by nosheenSheikh02 — APPROVED** [review]

**Pull request merged**